packages/cli/src/index.ts

```diff
@@ -0,0 +1,152 @@
+    // .option("-t, --template <template-name>", "Supported Semaphore template.", "hardhat")
+    .action(async (projectDirectory) => {
+        const currentDirectory = process.cwd()
+        const spinner = new Spinner(`Creating your project in ${chalk.green(`./${projectDirectory}`)}`)
+        const templateURL = `https://registry.npmjs.org/@semaphore-protocol/cli-template-hardhat/-/cli-template-hardhat-${version}.tgz`
+
+        if (existsSync(projectDirectory)) {
+            console.info(`\n ${logSymbols.error}`, `error: the '${projectDirectory}' folder already exists\n`)
+            return
+        }
+
+        spinner.start()
+
+        await download(templateURL, currentDirectory, { extract: true })
+
+        renameSync(`${currentDirectory}/package`, `${currentDirectory}/${projectDirectory}`)
+
+        spinner.stop()
+
+        console.info(` ${logSymbols.success}`, `Your project is ready!\n`)
+        console.info(` Please, install your dependencies by running:\n`)
+        console.info(`   ${chalk.cyan("cd")} ${projectDirectory}`)
+        console.info(`   ${chalk.cyan("npm i")}\n`)
+
+        const { scripts } = JSON.parse(readFileSync(`${currentDirectory}/${projectDirectory}/package.json`, "utf8"))
+
+        if (scripts) {
+            console.info(` Available scripts:\n`)
+
+            console.info(
+                `${Object.keys(scripts)
+                    .map((s) => `   ${chalk.cyan(`npm run ${s}`)}`)
+                    .join("\n")}\n`
+            )
+        }
+    })
+
+program
+    .command("get-groups")
+    .description("Get the list of groups from a supported network (Goerli or Arbitrum).")
+    .option("-n, --network <network-name>", "Supported Ethereum network.", "goerli")
+    .action(async ({ network }) => {
+        const subgraph = new Subgraph(network)
+        const spinner = new Spinner("Fetching groups")
+
+        spinner.start()
+
+        try {
+            const groups = await subgraph.getGroups()
+
+            spinner.stop()
+
+            if (groups.length === 0) {
+                console.info(` ${logSymbols.error}`, "error: there are no groups in this network\n")
+                return
+            }
+
+            const content = `${groups.map(({ id }: any) => ` - ${id}`).join("\n")}`
+
+            console.info(`${content}\n`)
+        } catch (error) {
+            spinner.stop()
+
+            console.info(` ${logSymbols.error}`, "error: unexpected error with the Semaphore subgraph")
+        }
+    })
+
+program
+    .command("get-group")
+    .description("Get the data of a group from a supported network (Goerli or Arbitrum).")
+    .argument("<group-id>", "Identifier of the group.")
+    .option("-n, --network <network-name>", "Supported Ethereum network.", "goerli")
+    .option("--members", "Show group members.")
+    .option("--signals", "Show group signals.")
+    .action(async (groupId, { network, members, signals }) => {
+        const subgraph = new Subgraph(network)
```

Comment:
same check that network is valid?

For `groupId`, I am not sure a select makes sense... too many values to choose from

Reply:
Too many values yep, there could be thousands